.automation/copy-issue-labels-to-pr

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: Apache-2.0
+
+import itertools
+import json
+import re
+import sys
+from github import Github
+
+# We don't want to copy all labels on linked issues; only those in this subset.
+COPYABLE_LABELS = {
+    "amd sev",
+    "documentation",
+    "ibm pef",
+    "infrastructure",
+    "intel sgx",
+    "wasm",
+}
+
+# Get inputs from shell
+(token, repository, path) = sys.argv[1:4]
+
+# Authenticate with Github using our token
+g = Github(token)
+
+# Initialize repo
+repo = g.get_repo(repository)
+
+# Open Github event JSON
+with open(path) as f:
+    event = json.load(f)
+
+# --- Extract PR from event JSON ---
+# `check_run`/`check_suite` does not include any direct reference to the PR
+# that triggered it in its event JSON. We have to extrapolate it using the head
+# SHA that *is* there.
+
+# Get head SHA from event JSON
+pr_head_sha = event["check_suite"]["head_sha"]
+
+# Find the repo PR that matches the head SHA we found
+pr = {pr.head.sha: pr for pr in repo.get_pulls()}[pr_head_sha]
+
+# Extract all associated issues from closing keyword in PR
+regex_keywords = re.compile("(close[sd]?|fix|fixe[sd]?|resolve[sd]?)\s*:?\s+#(\d+)", re.I)
+closing_numbers_pr_body = {number for keyword, number in regex_keywords.findall(pr.body)}
+
+# Extract all associated issues from PR commit messages
+results_commit_messages = [regex_keywords.findall(c.commit.message) for c in pr.get_commits()]
+closing_numbers_commit_messages = {num for verb, num in itertools.chain(*results_commit_messages)}
+
+# Get the union of both sets of associated issue numbers
+closing_numbers = closing_numbers_pr_body.union(closing_numbers_commit_messages)
+
+# If there are no associated issues, we're done. If not, we need to find out
+# which labels we need to add.
+if len(closing_numbers) == 0:
+    quit()
```

Comment:
I think you are quitting at the wrong place. We don't want to quit if there are no associated issues but if there are no labels to copy.

Reply:
My original thinking was that it would cut down on unnecessary API requests, and possibly avoid errors being thrown.

That said, I tried a version of the bot that didn't quit here on a PR that didn't link to anything, and it didn't fail. It also seems like it shouldn't make any additional API requests. So I can remove this. Probably best for tidiness.